CondFormats/BeamSpotObjects/interface/SimBeamSpotHLLHCObjects.h

```diff
@@ -0,0 +1,106 @@
+    fMeanZ = 0.0;
+    fEProton = 0.0;
+    fCrabFrequency = 0.0;
+    fRF800 = 0.0;
+    fCrossingAngle = 0.0;
+    fCrabbingAngleCrossing = 0.0;
+    fCrabbingAngleSeparation = 0.0;
+    fBetaCrossingPlane = 0.0;
+    fBetaSeparationPlane = 0.0;
+    fHorizontalEmittance = 0.0;
+    fVerticalEmittance = 0.0;
+    fBunchLength = 0.0;
+    fTimeOffset = 0.0;
+  };
+
+  virtual ~SimBeamSpotHLLHCObjects(){};
+
+  /// set meanX, meanY, meanZ
+  void setMeanX(double val) { fMeanX = val; }
+  void setMeanY(double val) { fMeanY = val; }
+  void setMeanZ(double val) { fMeanZ = val; }
+  /// set EProton, fCrabFrequency, RF800
+  void setEProton(double val) { fEProton = val; }
+  void setCrabFrequency(double val) { fCrabFrequency = val; }
+  void setRF800(double val) { fRF800 = val; }
+  /// set Crossing and Crabbing angles
+  void setCrossingAngle(double val) { fCrossingAngle = val; }
+  void setCrabbingAngleCrossing(double val) { fCrabbingAngleCrossing = val; }
+  void setCrabbingAngleSeparation(double val) { fCrabbingAngleSeparation = val; }
+  /// set BetaStar and Emittance
+  void setBetaCrossingPlane(double val) { fBetaCrossingPlane = val; }
+  void setBetaSeparationPlane(double val) { fBetaSeparationPlane = val; }
+  void setHorizontalEmittance(double val) { fHorizontalEmittance = val; }
+  void setVerticalEmittance(double val) { fVerticalEmittance = val; }
+  /// set BunchLength and TimeOffset
+  void setBunchLength(double val) { fBunchLength = val; }
+  void setTimeOffset(double val) { fTimeOffset = val; }
+
+  /// get meanX, meanY, meanZ position
+  double meanX() const { return fMeanX; }
+  double meanY() const { return fMeanY; }
+  double meanZ() const { return fMeanZ; }
+  /// get EProton, fCrabFrequency, RF800
+  double eProton() const { return fEProton; }
+  double crabFrequency() const { return fCrabFrequency; }
+  double rf800() const { return fRF800; }
+  /// set Crossing and Crabbing angles
+  double crossingAngle() const { return fCrossingAngle; }
+  double crabbingAngleCrossing() const { return fCrabbingAngleCrossing; }
+  double crabbingAngleSeparation() const { return fCrabbingAngleSeparation; }
+  /// get BetaStar and Emittance
+  double betaCrossingPlane() const { return fBetaCrossingPlane; }
+  double betaSeparationPlane() const { return fBetaSeparationPlane; }
+  double horizontalEmittance() const { return fHorizontalEmittance; }
+  double verticalEmittance() const { return fVerticalEmittance; }
+  /// get BunchLength and TimeOffset
+  double bunchLenght() const { return fBunchLength; }
+  double timeOffset() const { return fTimeOffset; }
+
+  /// print sim beam spot parameters
+  void print(std::stringstream& ss) const;
+
+private:
+  double fMeanX, fMeanY, fMeanZ;
+  double fEProton, fCrabFrequency, fRF800;
+  double fCrossingAngle, fCrabbingAngleCrossing, fCrabbingAngleSeparation;
+  double fBetaCrossingPlane, fBetaSeparationPlane;
+  double fHorizontalEmittance, fVerticalEmittance;
+  double fBunchLength, fTimeOffset;
```

Comment:
I think the main point of this PR is if this set of variables is exhaustive or not (in other words if `HLLHCEvtVtxGenerator` is susceptible of being changed later on. I am not sure who is in the best position to answer this question, but let me tag @cms-sw/simulation-l2 in case they have an idea.

Reply:
Thanks Marco I totally agree!
This point was also raised in:
 - https://github.com/cms-sw/cmssw/pull/43160#issue-1971055726
 - https://github.com/cms-AlCaDB/AlCaTools/issues/86#issuecomment-1781503058

but I got no answer so far...
Let me tag also @lgray who last worked on the `HLLHCEvtVtxGenerator` plugin.

Reply:
A kind ping @cms-sw/simulation-l2 @cms-sw/upgrade-l2 @lgray

Reply:
It looks to me that this PR needs to be discussed outside of a PR thread, given there is a lack of responses.
Perhaps a talk at some upcoming simulation meeting should be scheduled (hopefully capturing the relevant attendance).

Reply:
Possible for this week SIM meeting (Friday, 15:30) ?

Reply:
By the way, I am not sure if simulation, or upgrade are the contact persons of these parameters (at least until now). Our side is more on the software. For the conditions, we get them somehow by requests which may point to discussion.

Reply:
For reference:
This development was presented and discussed at the [Simulation meeting on May 24, 2024](https://indico.cern.ch/event/1419327/#39-sim-beam-spot-hllhc-condfor) and it was eventually decided to proceed with the integration of this PR as is, see https://github.com/cms-sw/cmssw/pull/43186#issuecomment-2190998051.